### ISSUE TITLE
JavaScript heap out of memory when building tenant registration container image

### ISSUE THREAD

**Issue by lspellman** [aws-samples/aws-saas-factory-eks-saas-workshop#4]
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory

This occurs when executing the instructions in the workshop for:
Step 9 - Build and deploy tenant registration container image:

```
cd ~/environment/aws-saas-factory-eks-saas-workshop/
./scripts/deploy-tenant-registration.sh
```

The context is a Cloud9 environment for the workshop with an m5.large instance with 8GB memory.

It happens at this point in the process of the Dockerfile instructions:

```
Step 6/11 : RUN yarn build tenant-registration
 ---> Running in 842e994ad281
yarn run v1.22.10
$ rimraf dist
$ nest build tenant-registration
```

**Comment by tobuck-aws** [comment]
I just added a yarn.lock file which may assist with this. Can you re-pull the repo and give it a shot?

**Comment by lspellman** [comment]
Thanks for that. A couple more changes were needed in Dockerfile.tenant-registration to get it to work.

first, update the node version on line 1, because the build fails with older node.

_error @angular-devkit/core@13.1.2: The engine "node" is incompatible with this module. Expected version "^12.20.0 || ^14.15.0 || >=16.10.0". Got "16.5.0"
error Found incompatible module._

`FROM public.ecr.aws/bitnami/node:16.13.1 AS build`

That will be needed again in the second section of the file. 

Then, with the new yarn.lock file, it also needs to be copied into the /app directory along with package.json, so I added this line:

`COPY yarn.lock ./`

Here's the full content of the updated Dockerfile.tenant-registration file:

FROM public.ecr.aws/bitnami/node:16.13.1 AS build
WORKDIR /app
COPY package.json ./
COPY yarn.lock ./
RUN yarn
COPY . .
RUN yarn build tenant-registration

FROM public.ecr.aws/bitnami/node:16.13.1
WORKDIR /app
COPY --from=build /app ./
EXPOSE 3000
CMD ["npm", "run", "start:tr"]

With that, I was able to get a successful image created and uploaded to the container. I suspect I'll need to do this later as well with the other services like user-management. I'll see.
